### PR TITLE
[SPARKR-195] Data frames now coerced to lists, and messages issued for a data frame o...

### PR DESCRIPTION
**Pull request by dputler**
...r matrix on how they are parallelized

### CODE REVIEW

**Comment by dputler** [comment]
Adding this task to the Jira as [SparkR-195]

File: pkg/R/context.R [hunk 1]
```diff
@@ -85,11 +85,14 @@ parallelize <- function(sc, coll, numSlices = 1) {
   # TODO: bound/safeguard numSlices
   # TODO: unit tests for if the split works for all primitives
   # TODO: support matrix, data frame, etc
-  if (!is.list(coll)) {
-    if (!is.vector(coll)) {
+  if ((!is.list(coll) && !is.vector(coll)) || is.data.frame(coll)) {
+    if (is.data.frame(coll))
```
Comment (shivaram) [inline pkg/R/context.R]: Minor style nit: We generally use `{` and `}` in SparkR even if we have one line statements.

**Comment by shivaram** [comment]
Thanks @dputler for the change. LGTM, but for the minor style comment.

**Comment by dputler** [comment]
I'll make the change, I actually prefer since it makes the code easier to follow. However, and FYI, R Core Team members argue for minimizing the number of curly braces, since each is a function call, and has overhead implications. The change will come as a supplementary pull request.

**Comment by dputler** [comment]
The style guideline changes have been implemented.

**Comment by shivaram** [comment]
Thanks @dputler - Merged this.

**Pull request merged**